File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/notification/NotificationContract.java
package org.mifos.mobilewallet.mifospay.notification;

import org.mifos.mobilewallet.core.domain.model.NotificationPayload;
import org.mifos.mobilewallet.mifospay.base.BasePresenter;
import org.mifos.mobilewallet.mifospay.base.BaseView;

import java.util.List;

/**
 * This is a contract class working as an Interface for UI
 * and Presenter components of the Notification package.
 * @author ankur
 * @since 24/July/2018
 */

public interface NotificationContract {

    /**
     * Defines all the functions in Presenter component.
     */
    interface NotificationPresenter extends BasePresenter {

        void fetchNotifications();
    }

    /**
     * Defines all the functions in UI component.
     */
    interface NotificationView extends BaseView<NotificationPresenter> {

        void fetchNotificationsSuccess(List<NotificationPayload> notificationPayloadList);

        void fetchNotificationsError(String message);
    }
}

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/notification/NotificationContract.kt
package org.mifos.mobilewallet.mifospay.notification

import com.mifos.mobilewallet.model.domain.NotificationPayload
import org.mifos.mobilewallet.mifospay.base.BasePresenter
import org.mifos.mobilewallet.mifospay.base.BaseView

/**
 * Created by ankur on 24/July/2018
 */
interface NotificationContract {
    interface NotificationPresenter : BasePresenter {
        fun fetchNotifications()
    }

    interface NotificationView : BaseView<NotificationPresenter?> {
        fun fetchNotificationsSuccess(notificationPayloadList: List<NotificationPayload?>?)
        fun fetchNotificationsError(message: String?)
    }
}